src/main/java/uk/gov/companieshouse/api/accounts/service/impl/AbstractServiceImpl.java

```diff
@@ -37,23 +43,32 @@ public AbstractServiceImpl(MongoRepository<U, String> mongoRepository,
     }
 
     @Override
-    public ResponseObject<T> create(T rest, String companyAccountId) {
+    public ResponseObject<T> create(T rest, Transaction transaction, String requestId)
+            throws DataException {
+        String companyNumber = transaction.getCompanyNumber();
+
         addEtag(rest);
         addKind(rest);
         U baseEntity = genericTransformer.transform(rest);
 
-        try{
-        baseEntity.setId(generateID(companyAccountId));
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("transaction_id", transaction.getId());
+        debugMap.put("company_accounts_id", companyNumber);
+
+        try {
+            baseEntity.setId(generateID(companyNumber));
```

Comment:
should be using company accounts id not company number

Reply:
generateID needs to be different per resource, it is built up from `companyaccountsId` concatentated with the resource type so we'd need the resource type either passed as a parameter or different implementations of generateID per resource

e.g. for a small-full resource we'd use a hash of `{companyAccountId}-small-full` and for a current period we'd use `{companyAccountId}-current-period`